src/frontend/src/lib/flows/startAddPasskeyFlow.svelte.ts

```diff
@@ -0,0 +1,62 @@
+import { DeviceData, UserNumber } from "$lib/generated/internet_identity_types";
+import { authenticatedStore } from "$lib/stores/authentication.store";
+import { waitFor } from "$lib/utils/utils";
+import { nonNullish } from "@dfinity/utils";
+import { get } from "svelte/store";
+
+export class StartAddPasskeyFlow {
+  view = $state<"show-link" | "authorize" | "success">("show-link");
+  #identityNumber: UserNumber;
+  #tentativeDevice: DeviceData | undefined;
+
+  constructor(identityNumber: UserNumber) {
+    this.#identityNumber = identityNumber;
+  }
+
+  startAddPasskeyFlow = async () => {
+    console.log("starting add passkey flow");
+    await this.#enterRegistrationMode();
+    console.log("registration mode entered");
+    this.#tentativeDevice = await this.#pollForTentativeDevice();
+    console.log(this.#tentativeDevice);
+    this.view = "authorize";
+  };
+
+  verifyDevice = async (code: string) => {
+    console.log("verifying device");
+    const actor = await get(authenticatedStore).actor;
+    const response = await actor.verify_tentative_device(
+      this.#identityNumber,
+      code,
+    );
+    console.log(response);
+    if ("verified" in response) {
+      this.view = "success";
+    }
+  };
+
+  #enterRegistrationMode = async () => {
+    const actor = await get(authenticatedStore).actor;
+    await actor.enter_device_registration_mode(this.#identityNumber);
+  };
+
+  #pollForTentativeDevice = async (): Promise<DeviceData | undefined> => {
+    const actor = await get(authenticatedStore).actor;
+    // eslint-disable-next-line no-async-promise-executor
+    return new Promise<DeviceData | undefined>(async (resolve) => {
+      // TODO: Add countdown or timeout
+      while (true) {
```

Comment:
I added it for the PoC, but I don't think we should have a `while (true)` in production 😂

Reply:
Just add a simple counter of calls for now and add the time counter in another PR.